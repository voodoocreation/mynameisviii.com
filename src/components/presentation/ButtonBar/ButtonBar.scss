@import "../../../scss/variables", "../../../scss/mixins";

.ButtonBar {
  display: flex;
  align-items: center;
  margin-bottom: $spacing;

  &::before,
  &::after {
    display: block;
    content: "";
    flex-grow: 1;
    height: 1px;
    background: $text-color;
  }

  .Button {
    margin: 0 $spacing-half;
  }

  @media print {
    &::before,
    &::after {
      height: 0;
      border-top: #000 1px solid;
      background: none;
    }
  }
}
