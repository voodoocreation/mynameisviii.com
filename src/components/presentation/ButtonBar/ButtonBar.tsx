import cn from "classnames";
import * as React from "react";

import "./ButtonBar.scss";

interface IProps {
  className?: string;
}

const ButtonBar: React.FC<IProps> = ({ children, className }) =>
  !children ? null : (
    <div className={cn("ButtonBar", className)}>{children}</div>
  );

export default ButtonBar;
