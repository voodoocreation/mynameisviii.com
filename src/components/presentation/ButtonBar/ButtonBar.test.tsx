import * as React from "react";

import WrapperWithIntl from "../../../utilities/WrapperWithIntl";
import ButtonBar from "./ButtonBar";

const component = new WrapperWithIntl(ButtonBar);

describe("[presentation] <ButtonBar />", () => {
  it("doesn't render anything when there are no children", () => {
    const wrapper = component.render();

    expect(wrapper.html()).toBeNull();
  });

  describe("when there are children", () => {
    const wrapper = component.withChildren(<div className="Child" />).render();

    it("renders the children", () => {
      expect(wrapper.find(".Child")).toHaveLength(1);
    });

    it("matches snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
