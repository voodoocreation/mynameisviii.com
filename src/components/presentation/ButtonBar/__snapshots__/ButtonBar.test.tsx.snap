// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[presentation] <ButtonBar /> when there are children matches snapshot 1`] = `
<div
  class="ButtonBar"
>
  <div
    class="Child"
  />
</div>
`;
